components/sidebar/sidebar_channel/sidebar_group_channel/sidebar_group_channel.tsx

```diff
@@ -2,25 +2,65 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {Link} from 'react-router-dom';
 
 import {Channel} from 'mattermost-redux/types/channels';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import {trackEvent} from 'actions/diagnostics_actions';
+import {browserHistory} from 'utils/browser_history';
+import Constants from 'utils/constants';
+
+import SidebarChannelLink from 'components/sidebar/sidebar_channel/sidebar_channel_link';
 
 type Props = {
     channel: Channel;
     currentTeamName: string;
+    currentUserId: string;
+    redirectChannel: string;
+    active: boolean;
+    membersCount: number;
+    actions: {
+        savePreferences: (userId: string, preferences: PreferenceType[]) => Promise<{data: boolean}>;
+    };
 };
 
 type State = {
 
 };
 
 export default class SidebarGroupChannel extends React.PureComponent<Props, State> {
+    handleLeaveChannel = (callback: () => void) => {
+        const id = this.props.channel.id;
+        const category = Constants.Preferences.CATEGORY_GROUP_CHANNEL_SHOW;
+
+        const currentUserId = this.props.currentUserId;
+        this.props.actions.savePreferences(currentUserId, [{user_id: currentUserId, category, name: id!, value: 'false'}]).then(
+            () => callback);
+
+        trackEvent('ui', 'ui_direct_channel_x_button_clicked');
+
+        if (this.props.active) {
+            browserHistory.push(`/${this.props.currentTeamName}/channels/${this.props.redirectChannel}`);
+        }
+    }
+
+    getIcon = () => {
+        return (
+            <div className='status status--group'>{this.props.membersCount}</div>
```

Comment:
Relating to the above comment about the icon components, this could be another `GroupMessageIcon` component

Reply:
I think localizing logic by channel type splits things out enough.